File: RetrofitTwoProject/app/src/main/java/com/calvert/retrofittwoproject/MainActivity.java
package com.calvert.retrofittwoproject;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import com.calvert.retrofittwoproject.adapter.RecyclerViewAdapter;
import com.calvert.retrofittwoproject.model.Comments;
import com.calvert.retrofittwoproject.network.ApiInterface;
import com.calvert.retrofittwoproject.network.RetrofitInstance;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = MainActivity.class.getSimpleName();

    private RecyclerView recyclerView;
    private RecyclerViewAdapter recyclerViewAdapter;
    private RecyclerView.LayoutManager layoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerview);
        layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);


        ApiInterface apiInterface = RetrofitInstance.getRetrofitInstance().create(ApiInterface.class);
        Call<List<Comments>> call = apiInterface.getComments();

        call.enqueue(new Callback<List<Comments>>() {
            @Override
            public void onResponse(Call<List<Comments>> call, Response<List<Comments>> response) {
                Log.d(TAG, "onResponse: " + response.body().get(0).getName());
                recyclerViewAdapter = new RecyclerViewAdapter(response.body(), MainActivity.this);
                recyclerView.setAdapter(recyclerViewAdapter);
            }

            @Override
            public void onFailure(Call<List<Comments>> call, Throwable t) {
                Log.d(TAG, "onFailure: " + t.getMessage());
            }
        });
    }
}

File: KotlinRetrofitRxJavaRecyclerView3/app/src/main/java/com/calvert/kotlinretrofitrxjavarecyclerview3/MainActivity.kt
package com.calvert.kotlinretrofitrxjavarecyclerview3

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.util.Log
import com.calvert.kotlinretrofitrxjavarecyclerview3.adapter.MyAdapter
import com.calvert.kotlinretrofitrxjavarecyclerview3.model.Comment
import com.calvert.kotlinretrofitrxjavarecyclerview3.network.ApiInterface
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    //init later using lateinit
    lateinit var recyclerView: RecyclerView
    lateinit var myAdapter: MyAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recyclerview)

        recyclerView.layoutManager = LinearLayoutManager(this)

        val apiInterface = ApiInterface.getRetrofit().getPost()
        apiInterface.enqueue(object: Callback<List<Comment>> {
            override fun onFailure(call: Call<List<Comment>>, t: Throwable) {
                Log.e("Error", t.message)
            }

            override fun onResponse(call: Call<List<Comment>>, response: Response<List<Comment>>) {
                //!! means not null type
                myAdapter = MyAdapter(this@MainActivity, response.body()!!)
                recyclerView.adapter = myAdapter
                myAdapter.notifyDataSetChanged()
            }
        })
    }
}

